File: python/pig-latin/pig_latin.py
import re
def translate(text):
    out = []
    words = text.split(' ')
    for word in words:
        if re.match("^(xr|yr|yt|[aeiou])", word):
            out.append(word + 'ay')
            continue

        m = re.match("^(.?qu)(.*)$", word)
        if m:
            out.append(m.group(2) + m.group(1) + 'ay')
            continue

        m = re.search("^([^aeiou]+?)([aeiouy].*)$", word)
        if m:
            out.append(m.group(2) + m.group(1) + 'ay')
            continue

    return ' '.join(out)

File: ruby/pig-latin/pig_latin.rb
class PigLatin
  def self.translate(words)
    words.split.map { |word|
      word.match?(/^([aieou]|xr|yt)/) || word.sub!(/^(.?qu|[^aeiou]+(?=y)|[^aeiou]+)(.*?)$/, '\2\1')
      word + "ay"
    }.join(" ")
  end
end
